faw/pdf-observatory/faw_analysis_set_parse.py

```diff
@@ -270,22 +282,21 @@ def _dask_as_parse_file(app_config, api_info, doc):
             raise ValueError("Could not find idle?")
         doc['parsers'].update(idle_parsers_doc['parser_versions'][1])
 
-    def get_cfg(k):
-        """Given a parser name `k`, resolve to relevant app_config"""
-        parser_pipeline_name = faw_analysis_set_util.deconstruct_pipeline_parser_name(k)
-        if parser_pipeline_name is None:
-            parser_config = app_config['parsers'][k]
-        else:
-            parser_config = (app_config['pipelines']
-                    [parser_pipeline_name.pipe]
-                    ['parsers']
-                    [parser_pipeline_name.parser])
-            parser_config = parser_config.copy()
-            parser_config['aset'] = parser_pipeline_name.aset
-            parser_config['pipeline'] = parser_pipeline_name.pipe
-        return parser_config
-
     parser_set = list(doc.get('parsers', {}).keys())
+    # parser_set may not include all upstream/downstream parsers to run
+    all_parser_configs = faw_analysis_set_util.lookup_all_parsers(
```

Comment:
I'm using `faw_analysis_set_util.lookup_all_parsers` to get parser configs in a few places, since we need configs from (upstream) parsers outside the analysis set that aren't in `doc['parsers']`. I think this is the best way to just "get all the parser configs", but let me know if there's a better one that hits the DB less.

Reply:
Not that I'm aware of; that said, as you appear to be doing, cache that response where possible since it does have to ask the DB about pipeline plugins.

Reply:
There is one more opportunity for caching this value that I didn't take:

Currently, this is called once per file when rerunning parsers, to get configs for upstream parsers from other analysis sets. If upstream parsers were marked for rerun in `as_parse`, we wouldn't need to hit the db for configs in `faw_analysis_set_parse`. I didn't mark them for rerun since we don't strictly need them to be rerun (we only need their artifacts). Marking the upstream parsers for rerun won't do anything if their versions didn't change and they're no longer upstream from anything, though, so it wouldn't hurt to do so. I'll make this change.